File: src/core/jupiter/core/common/sub/persons/namespace.py
"""The namespace of a person link."""

from jupiter.framework.value import EnumValue, enum_value


@enum_value
class PersonNamespace(EnumValue):
    """The namespace of a person link."""

    PRM = "prm"
    INBOX_TASK = "inbox-task"
    BIG_PLAN = "big-plan"
    SCHEDULE = "schedule"
    HABIT = "habit"
    CHORE = "chore"
    SMART_LIST_ITEM = "smart-list-item"
    METRIC_ENTRY = "metric-entry"

File: src/core/jupiter/core/common/sub/persons/namespace.ts
import { PersonNamespace } from "@jupiter/webapi-client";

export function personNamespaceName(namespace: PersonNamespace): string {
  switch (namespace) {
    case PersonNamespace.PRM:
      return "PRM";
    case PersonNamespace.INBOX_TASK:
      return "Inbox Task";
    case PersonNamespace.BIG_PLAN:
      return "Big Plan";
    case PersonNamespace.SCHEDULE:
      return "Schedule";
    case PersonNamespace.HABIT:
      return "Habit";
    case PersonNamespace.CHORE:
      return "Chore";
    case PersonNamespace.SMART_LIST_ITEM:
      return "Smart List Item";
    case PersonNamespace.METRIC_ENTRY:
      return "Metric Entry";
  }
}
